File: java/tw/edu/au/csie/ucan/beebit/dec.java
package tw.edu.au.csie.ucan.beebit;

class dec {
	public static void main(String[] args) {

		if(args.length < 4) {
			System.out.printf("Usage: dec [pk] [sk] [ct] [pt]\n");
			System.out.printf("pk: path to public key\n");
			System.out.printf("sk: path to secret key\n");
			System.out.printf("ct: path to encrypted file (cipher text)\n");
			System.out.printf("pt: path to original file (plain text)\n");
			System.exit(-1);
		}

		String pk = args[0];
		String sk = args[1];
		String ct = args[2];
		String pt = args[3];

		cpabeJNI bee = new cpabeJNI();

		int len = bee.fdec(pk, sk, ct, pt);
		if(len == -1) {
			System.out.printf("Decrypt failed!\n");
			System.exit(-1);
		}
		System.out.printf("Decrypt success!\n");
		System.out.printf("File (%s) decrypted to file (%s).\n", ct, pt);
	}	
}

File: python/dec.py
import sys
import pycpabe

argc = len(sys.argv)

if argc < 5:
	print("Usage: dec [pk] [sk] [ct] [pt]")
	print("pk: path to public key")
	print("sk: path to secret key")
	print("ct: path to encrypted file (cipher text)")
	print("pt: path to original file (plain text)")
	sys.exit()

pk = sys.argv[1]
sk = sys.argv[2]
ct = sys.argv[3]
pt = sys.argv[4]

if pycpabe.cpabe_fdec(pk, sk, ct, pt) == -1:
	print("Decrypt failed!")
	sys.exit()

print("Decrypt success!")
print("File (%s) decrypted to file (%s) " %(ct, pt))
